### PR TITLE
Bdd testing update customer

### PR DESCRIPTION
**Pull request by mwj10**
Added address id to UI + modified rest_api.js to have function that takes two responses for two AJAX calls + added update scenarios for update in the feature file

### CODE REVIEW

File: features/customers.feature [hunk 1]
```diff
@@ -159,4 +159,107 @@ Scenario: Create a Customer with Missing Password
     And I should see "Required field" in the "Password" error string
     When I set the "Password" to "test"
     And I press the "Create" button
+    Then The "Password" error string should be gone
+
+Scenario: Update a Customer
+    When I visit the "Home Page"
+    And I press the "Search" button
+    Then I should see the message "Success"
+    When I copy the "id" field
```
Comment (Akshama) [inline features/customers.feature:168]: ~id~ Id

File: features/customers.feature [hunk 2]
```diff
@@ -159,4 +159,107 @@ Scenario: Create a Customer with Missing Password
...
+    And I set the "First Name" to "Test FN"
+    And I set the "Last Name" to "Test LN"
+    And I set the "Email" to "test@gmail.com"
+    And I set the "Password" to "test788pass"
+    And I select "True" in the "Active" dropdown
+    And I set the "street" to "10th Casselberry Way"
```
Comment (Akshama) [inline features/customers.feature:174]: ~street~ Street
Similarly, Can you make all of the fields and button names to be PascalCase with first letter capitalized?

File: service/static/js/rest_api.js [hunk 3]
```diff
@@ -404,6 +454,7 @@ $(function () {
             table += '<th class="col-md-2">Last Name</th>'
             table += '<th class="col-md-2">Email</th>'
             table += '<th class="col-md-2">Password</th>'
+            table += '<th class="col-md-2">Add_Id</th>'
```
Comment (Akshama) [inline service/static/js/rest_api.js:457]: Add_ID just to be consistent with Cust_ID.

**Review by Akshama — CHANGES_REQUESTED** [review]

File: service/static/js/rest_api.js [hunk 4]
```diff
@@ -11,18 +11,41 @@ $(function () {
...
     }
 
+    // Updates the form with data from two responses
+    function update_form_data_two_responses(res1,res2) {
+        $("#customer_id").val(res1.id);
+        $("#customer_first_name").val(res1.first_name);
+        $("#customer_last_name").val(res1.last_name);
+        $("#customer_email").val(res1.email);
+        $("#customer_password").val(res1.password);
+        $("#customer_address_id").val(res2.address_id);
```
Comment (Akshama) [inline service/static/js/rest_api.js:35]: displayFieldRequiredNotification if address_id is missing.

**Review by Akshama — CHANGES_REQUESTED** [review]

File: features/customers.feature (hunk 1 above)
Comment (mwj10) [reply features/customers.feature]: fixed

File: features/customers.feature (hunk 2 above)
Comment (mwj10) [reply features/customers.feature]: fixed

File: service/static/js/rest_api.js (hunk 3 above)
Comment (mwj10) [reply service/static/js/rest_api.js:539]: Fixed

File: service/static/js/rest_api.js (hunk 4 above)
Comment (mwj10) [reply service/static/js/rest_api.js:35]: added

File: features/customers.feature [hunk 5]
```diff
@@ -174,4 +174,107 @@ Scenario: Create a Customer with Missing Password
...
+    And I set the "City" to "monroe"
+    And I set the "Pin Code" to " 07319"
+    And I set the "State" to "NJ"
+    And I set the "Country" to "USA"
+    And I press the "update" button
+    Then I should see the message "Form Error(s)"
+    And I should see "Required field" in the "Password" error string
+    When I set the "Password" to "test"
+    And I press the "Update" button
+    Then The "Password" error string should be gonegit
```
Comment (Akshama) [inline features/customers.feature:280]: ~goneit~ gone

**Review by Akshama — CHANGES_REQUESTED** [review]

File: features/customers.feature (hunk 5 above)
Comment (mwj10) [reply features/customers.feature]: Fixed

File: features/customers.feature [hunk 6]
```diff
@@ -220,3 +220,114 @@ Scenario: Deactivate and Activate a Customer
...
+    And I set the "Pin Code" to " 07319"
+    And I set the "State" to "NJ"
+    And I set the "Country" to "USA"
+    And I press the "update" button
+    Then I should see the message "Success"
+    When I press the "clear" button
+    And I paste the "Id" field
+    And I press the "retrieve" button
+    Then I should see the message "Success"
+    And I should see "Test FN" in the "first_name" field
```
Comment (Akshama) [inline features/customers.feature:246]: Can you change the field names in this block as well according to UI?
~first_name~ First Name and so on.

File: features/customers.feature [hunk 7]
```diff
@@ -220,3 +220,114 @@ Scenario: Deactivate and Activate a Customer
...
+    And I set the "City" to "monroe"
+    And I set the "Pin Code" to " 07319"
+    And I set the "State" to "NJ"
+    And I set the "Country" to "USA"
+    And I press the "update" button
+    Then I should see the message "Form Error(s)"
+    And I should see "Required field" in the "Password" error string
+    When I set the "Password" to "test"
+    And I press the "Update" button
+    Then The "Password" error string should be gone
```
Comment (Akshama) [inline features/customers.feature:333]: Can we add the scenario for Updating with missing email - for completion.

Comment (mwj10) [reply features/customers.feature:352]: Done

**Comment by Akshama** [comment]
Everything is in order! looks good to me. Merging.

**Pull request merged**